Code/Framework/AzCore/AzCore/Memory/AllocatorManager.cpp

```diff
@@ -7,401 +7,882 @@
 namespace AZ
 {
+    static void sys_DumpAllocators([[maybe_unused]] const AZ::ConsoleCommandContainer& arguments)
+    {
+        AllocatorManager::Instance().DumpAllocators();
+    }
+    AZ_CONSOLEFREEFUNC(sys_DumpAllocators, AZ::ConsoleFunctorFlags::Null, "Print memory allocator statistics.");
 
-static void sys_DumpAllocators([[maybe_unused]] const AZ::ConsoleCommandContainer& arguments)
-{
-    AllocatorManager::Instance().DumpAllocators();
-}
-AZ_CONSOLEFREEFUNC(sys_DumpAllocators, AZ::ConsoleFunctorFlags::Null, "Print memory allocator statistics.");
+    // Provides a range of allocations to dump. The min value is inclusive and the max value is exclusive
+    // Theerefore the range is [min, max)
+    struct AllocationDumpRange
+    {
+        size_t m_min = 0;
+        size_t m_max = AZStd::numeric_limits<size_t>::max();
+    };
 
-static EnvironmentVariable<AllocatorManager>& GetAllocatorManagerEnvVar()
-{
-    static EnvironmentVariable<AllocatorManager> s_allocManager;
-    return s_allocManager;
-}
+    // Add console command for outputting the Allocation records for all or a specified set of allocators with the provided names
+    static void DumpAllocationsForAllocatorHelper(
+        AZStd::span<AZStd::string_view const> allocatorNameArguments,
+        AZ::IO::GenericStream& printStream,
+        const AllocationDumpRange& allocationDumpRange = {})
+    {
+        auto& allocatorManager = AZ::AllocatorManager::Instance();
+        const int numAllocators = allocatorManager.GetNumAllocators();
+        // Garbage collect the allocations before dumping them
+        allocatorManager.GarbageCollect();
 
-static AllocatorManager* s_allocManagerDebug;  // For easier viewing in crash dumps
+        AZStd::vector<IAllocator*> allocatorsToDump;
+        for (int i = 0; i < numAllocators; i++)
+        {
+            auto* allocator = allocatorManager.GetAllocator(i);
+            const AZ::Debug::AllocationRecords* records = allocator->GetRecords();
+            if (records)
+            {
+                // If no allocator name arguments to the Console command have been supplied
+                // dump all allocators that record allocation records
+                if (allocatorNameArguments.empty())
+                {
+                    allocatorsToDump.emplace_back(allocator);
+                    // Skip back to the next allocator
+                    continue;
+                }
 
-//////////////////////////////////////////////////////////////////////////
-bool AllocatorManager::IsReady()
-{
-    return GetAllocatorManagerEnvVar().IsConstructed();
-}
-//////////////////////////////////////////////////////////////////////////
-void AllocatorManager::Destroy()
-{
-}
+                // If the allocator has records and it matches the name of one of the allocators
+                auto IsAllocatorInNameSet = [allocator](AZStd::string_view searchName)
+                {
+                    return searchName == allocator->GetName();
```

Comment:
should we use a case-insensitive comparison here using "toLower"?

Reply:
I will change this to use `AZ::StringFunc::Equal` which does a case-insensitive compare